nixos/modules/services/misc/memos.nix

```
@@ -0,0 +1,197 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+let
+  cfg = config.services.memos;
+  envFileFormat = pkgs.formats.keyValue { };
+in
+{
+  options.services.memos = {
+    enable = lib.mkEnableOption "Memos note-taking";
+    package = lib.mkPackageOption pkgs "Memos" {
+      default = "memos";
+    };
+
+    openFirewall = lib.mkEnableOption "opening the ports in the firewall";
+
+    user = lib.mkOption {
+      type = lib.types.str;
+      description = ''
+        The user to run Memos as.
+
+        ::: {.note}
+        If changing the default value, **You** are responsible of creating the corresponding user with {option}`users.users`.
+        :::
+      '';
+      default = "memos";
+    };
+
+    group = lib.mkOption {
+      type = lib.types.str;
+      description = ''
+        The group to run Memos as.
+
+        ::: {.note}
+        If changing the default value, **You** are responsible of creating the corresponding group with {option}`users.groups`.
+        :::
+      '';
+      default = "memos";
+    };
+
+    dataDir = lib.mkOption {
+      default = "/var/lib/memos/";
+      type = lib.types.path;
+      description = ''
+        Specifies the directory where Memos will store its data.
+
+        ::: {.note}
+        It will be automatically created with the permissions of {option}`services.memos.user` and {option}`services.memos.group`.
+        :::
+      '';
+    };
+
+    settings = lib.mkOption {
+      type = envFileFormat.type;
+      description = ''
+        The environment variables to configure Memos.
+
+        ::: {.note}
+        At time of writing, there is no clear documentation about possible values.
+        It's possible to convert CLI flags into these variables.
+        Example : CLI flag "unix-sock" converts to {env}`MEMOS_UNIX_SOCK`.
+        :::
+      '';
+      example = {
+        MEMOS_MODE = "prod";
+        MEMOS_ADDR = "127.0.0.1";
+        MEMOS_PORT = "5230";
+        MEMOS_DATA = "/var/lib/memos/";
+        MEMOS_DRIVER = "sqlite";
+        MEMOS_INSTANCE_URL = "http://localhost:5230";
+      };
+      default = { };
+    };
+
+    environmentFile = lib.mkOption {
+      type = lib.types.path;
+      description = ''
+        The environment file to use when starting Memos.
+
+        ::: {.note}
+        By default, generated from {option}`services.memos.settings`.
+        :::
+      '';
+      example = "/var/lib/memos/memos.env";
+      default = envFileFormat.generate "memos.env" cfg.settings;
+      defaultText = ''
+        generated from {option}`services.memos.settings`
+      '';
```

1. We need `literalMD` here to render this value as markdown.
   When we don't use `literalMD` or `literalExpression`, the docs render the value using `lib.generators.toPretty`, so this would render as a nix string instead of raw markdown.
2. If we add a `opt = options.services.memos` to the file, we can interpolate options to get their actual attrpath.
   This makes things easier to maintain if options get renamed, etc.
3. We could consider a clickable link here, but that's unusual in default text IIUC.
   Not shown in my suggestion, it'd look like: `[](#opt-${opt.settings})`

```suggestion
      defaultText = lib.literalMD ''
        generated from {option}`${opt.settings}`
      '';
```

---

Nice ! Did not know this, I'm still not fluent in Nix option documentation and how It is rendered in the final docs. Thanks for clarifying